src/nvim/popupmenu.c

```diff
@@ -1191,7 +1202,9 @@ static bool pum_set_selected(int n, int repeat)
 
           if (use_float) {
             // adjust floating window by actually height and max info text width
-            pum_adjust_info_position(curwin, max_info_width);
+            if (!pum_adjust_info_position(curwin, max_info_width)) {
+              curwin = curwin_save;
```

Comment:
Setting `curwin` directly seems dubious. `curwin` after `win_close` in `pum_adjust_info_position` _should_ be `curwin_save` anyhow (makes this a no-op), but if that's somehow false then this is bugprone. (`curbuf` and other globals may remain out-of-sync)

Maybe use `win_enter`? (I'll assume `win_valid` isn't needed here as autocmds are blocked for popups... 😇)

```suggestion
              win_enter(curwin_save, false);
```